### DISCUSSION TITLE
Is there a way to get query params from page?

### DISCUSSION THREAD

**Discussion by zw963** [luckyframework/lucky#1883] [Q&A] [answered]
What i want to is, get query params, e.g.  params["q"], like "/universities?q=foo",and fill the `foo` into the search box when refresh the code.

Thanks

**Comment by jwoertink** [comment]
You can use `expose` for this case. https://luckyframework.org/guides/frontend/rendering-html#example-using-needs-and-expose

```crystal
class FooAction < BrowserAction
  param q : String
  expose query_param

  get "/universities" do
    html FooPage
  end

  private def query_param
    q
  end
end

class FooPage < MainLayout
  needs query_param : String

  def content
    h1 query_param
  end
end
```

**Reply by zw963** [reply, marked as answer]
Hi, i found i can get query_params use `context.request.query_params["q"]` in page directly,  this way is a little simple then pass from action,  furthermore, latter have to create a method.
